### PR TITLE
Feat/18308/ds popover header component

### PR DESCRIPTION
**Pull request by garrettbear**
## Explanation

Using `HeaderBase` build a `PopoverHeader` component that defaults with a back button in startAccessory position and close button in the endAccessory position. 

*Fixes #18308 

### After

Screenshot 2023-04-12 at 3 55 48 PM

## Manual Testing Steps
`yarn jest ui/components/component-library`

## Pre-merge author checklist

- [X] I've clearly explained:
  - [X] What problem this PR is solving
  - [X] How this problem was solved
  - [X] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [X] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [X] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

File: ui/components/component-library/popover-header/popover-header.stories.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  args: {
+    children: 'PopoverHeader',
+  },
+} as ComponentMeta<typeof PopoverHeader>;
+
+const Template: ComponentStory<typeof PopoverHeader> = (args) => {
+  return (
+    <PopoverHeader
+      onBack={() => console.log('back')}
+      onClose={() => console.log('close')}
```
Comment (georgewrmarshall) [inline ui/components/component-library/popover-header/popover-header.stories.tsx:24]: Can these be actions?

**Review by georgewrmarshall — COMMENTED** [review]
Looking good! Left a small suggestion

**Review by georgewrmarshall — COMMENTED** [review]
These should be resolved. They will fail the build if used in a .tsx file. The only reason it's not failing is because storybook files are currently being ignored but we will want to change that eventually

Screenshot 2023-04-18 at 7 18 08 PM
Screenshot 2023-04-18 at 7 18 15 PM

**Comment by georgewrmarshall** [comment]
> @georgewrmarshall do you remember what fix you had to do when running into the props not loading on README? Screenshot 2023-04-17 at 3 13 03 PM

Hmm I do not. Will check it out tomorrow

**Review by georgewrmarshall — APPROVED** [review]
LGTM! 🚀

File: ui/components/component-library/popover-header/popover-header.test.tsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
+/* eslint-disable jest/require-top-level-describe */
+import { render, fireEvent } from '@testing-library/react';
+import React from 'react';
+import { PopoverHeader } from './popover-header';
+
+describe('Popover', () => {
```
Comment (georgewrmarshall) [inline ui/components/component-library/popover-header/popover-header.test.tsx:6]: nit: Change to PopoverHeader
```suggestion
describe('PopoverHeader', () => {
```

File: ui/components/component-library/popover-header/popover-header.test.tsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  it('should render popover element correctly', () => {
```
Comment (georgewrmarshall) [inline ui/components/component-library/popover-header/popover-header.test.tsx:7]: nit
```suggestion
  it('should render popover header correctly', () => {
```

**Review by georgewrmarshall — DISMISSED** [review]
Had another review looks like we should still keep argTypes to prevent the stories from breaking. Controls don't really add any value if they don't work as expected. We may have to do some house keeping on the other TS components


https://user-images.githubusercontent.com/8112138/232970644-0723026f-a531-4d03-8cfe-ee1048810bcd.mov

**Review by NidhiKJha — DISMISSED** [review]

**Review by vthomas13 — APPROVED** [review]

**Review by NidhiKJha — APPROVED** [review]

**Pull request merged**